test/test_module_init.py

```diff
@@ -0,0 +1,330 @@
+        torch.nn.Threshold: ([0.1, 20], {}),
+        torch.nn.Transformer: ([], {}),
+        torch.nn.TransformerDecoder: ([torch.nn.TransformerDecoderLayer, 3], {}),
+        torch.nn.TransformerDecoderLayer: ([10, 2], {}),
+        torch.nn.TransformerEncoder: ([torch.nn.TransformerEncoderLayer, 3], {}),
+        torch.nn.TransformerEncoderLayer: ([10, 2], {}),
+        torch.nn.TripletMarginLoss: ([], {}),
+        torch.nn.TripletMarginWithDistanceLoss: ([], {}),
+        torch.nn.Unflatten: ([1, (2, 5, 5)], {}),
+        torch.nn.Unfold: ([3], {}),
+        torch.nn.Upsample: ([], {}),
+        torch.nn.UpsamplingBilinear2d: ([], {}),
+        torch.nn.UpsamplingNearest2d: ([], {}),
+        torch.nn.ZeroPad2d: ([0], {}),
+        torch.nn.qat.Conv2d: ([3, 3, 3], {'qconfig': torch.quantization.default_qconfig}),
+        torch.nn.qat.Conv3d: ([3, 3, 3], {'qconfig': torch.quantization.default_qconfig}),
+        torch.nn.qat.Linear: ([5, 2], {'qconfig': torch.quantization.default_qconfig}),
+        torch.nn.quantizable.LSTM: ([5, 6], {}),
+        torch.nn.quantizable.LSTMCell: ([5, 6], {}),
+        torch.nn.quantizable.MultiheadAttention: ([10, 2], {}),
+    }
+
+
+# Instantiates the given class with the given args, kwargs, optionally on a given device.
+def instantiate_class(cls, args, kwargs, device=None):
+    return cls(*args, **kwargs) if device is None else cls(*args, **kwargs, device=device)
+
+
+# Returns a function that calls the real implementation of a method
+# in addition to passing args to a mock object.
+def mock_wrapper(method):
+    mock = MagicMock()
+    def wrapper(self, *args, **kwargs):
+        mock(*args, **kwargs)
+        return method(self, *args, **kwargs)
+    wrapper.mock = mock
+    return wrapper
+
+
+# Returns a set of args / kwargs that can be used to construct the module.
+def get_example_args(module_cls, constructor_arg_db, device=None):
+    assert module_cls in constructor_arg_db, \
+        f"No entry for {module_cls} in the constructor arg DB. Please add it to pass these tests."
+    args, kwargs = constructor_arg_db[module_cls]
+
+    # Recursively instantiate args / kwargs that are class objects.
+    if device is None:
+        args = [instantiate_class(arg, *get_example_args(arg, constructor_arg_db))
+                if inspect.isclass(arg) else arg for arg in args]
+        kwargs = {k: instantiate_class(v, *get_example_args(v, constructor_arg_db))
+                  if inspect.isclass(v) else v for k, v in kwargs.items()}
+    else:
+        args = [instantiate_class(arg, *get_example_args(arg, constructor_arg_db), device=device)
+                if inspect.isclass(arg) else arg for arg in args]
+        kwargs = {k: instantiate_class(v, *get_example_args(v, constructor_arg_db), device=device)
+                  if inspect.isclass(v) else v for k, v in kwargs.items()}
+        kwargs['device'] = device
+    return args, kwargs
+
+
+def generate_tests(test_cls, constructor_arg_db):
```

Comment:
Add a comment describing what this test tests

Reply:
Test logic is actually below in `run_test()`, and its checks are commented

Reply:
I split the logic out for generating tests for a single module, so the separation between that and iterating over all the modules should be clearer now